cli/command/task/print.go

```diff
@@ -85,18 +85,7 @@ func print(out io.Writer, ctx context.Context, tasks []swarm.Task, resolver *idr
 		if err != nil {
 			return err
 		}
-
-		name := task.Annotations.Name
-		// TODO: This is the fallback <ServiceName>.<Slot>.<taskID> in case task name is not present in
-		// Annotations (upgraded from 1.12).
-		// We may be able to remove the following in the future.
-		if name == "" {
-			if task.Slot != 0 {
-				name = fmt.Sprintf("%v.%v.%v", serviceName, task.Slot, task.ID)
-			} else {
-				name = fmt.Sprintf("%v.%v.%v", serviceName, task.NodeID, task.ID)
-			}
-		}
+		name, err := resolver.Resolve(ctx, swarm.Task{}, task.ID)
```

Comment:
This adds an extra round-trip per task. Can you change `Resolve` to avoid the `TaskInspectWithRaw` request if `t.ID == id` (looks like it's usually called with an empty object).